test/integration/activate_int_test.go

```diff
@@ -551,3 +551,30 @@ func (suite *ActivateIntegrationTestSuite) TestActivateBranchNonExistant() {
 
 	cp.Expect("has no branch")
 }
+
+func (suite *ActivateIntegrationTestSuite) TestActivateArtifactsCached() {
```

Comment:
Suggest also testing that the cache is being used. You can do this by running with VERBOSE=true and asserting that the right debug logs happen (try not to be too specific in the assertions as debug messages tend to change over time).